package.json

```diff
@@ -51,140 +44,20 @@
   },
   "homepage": "https://github.com/goldcaddy77/warthog#readme",
   "//": "TODO: figure out which of these are dependencies, devDeps or peerDeps",
-  "dependencies": {
-    "@types/app-root-path": "^1.2.4",
-    "@types/caller": "^1.0.0",
-    "@types/cosmiconfig": "^5.0.3",
-    "@types/debug": "^0.0.31",
-    "@types/dotenv": "^6.1.1",
-    "@types/express": "^4.16.0",
-    "@types/graphql": "^14.1.1",
-    "@types/graphql-fields": "^1.3.0",
-    "@types/graphql-iso-date": "^3.3.1",
-    "@types/graphql-type-json": "^0.1.3",
-    "@types/isomorphic-fetch": "^0.0.34",
-    "@types/lodash": "^4.14.119",
-    "@types/mkdirp": "^0.5.2",
-    "@types/node": "^10.12.18",
-    "@types/node-emoji": "^1.8.0",
-    "@types/open": "^6.1.0",
-    "@types/pg": "^7.4.11",
-    "@types/prettier": "^1.15.2",
-    "@types/shortid": "^0.0.29",
-    "@types/ws": "^6.0.1",
-    "apollo-link-error": "^1.1.5",
-    "apollo-link-http": "^1.5.9",
-    "apollo-server": "^2.6.3",
-    "apollo-server-express": "^2.6.3",
-    "app-root-path": "^2.1.0",
-    "caller": "^1.0.1",
-    "class-transformer": "^0.2.0",
-    "class-validator": "^0.9.1",
-    "cosmiconfig": "^5.2.1",
-    "cross-fetch": "^3.0.0",
-    "dataloader": "^1.4.0",
-    "debug": "^4.1.1",
-    "express": "^4.16.4",
-    "gluegun": "^3.3.1",
-    "graphql": "^14.1.1",
-    "graphql-binding": "^2.5.1",
-    "graphql-fields": "^2.0.3",
-    "graphql-iso-date": "^3.6.1",
-    "graphql-tools": "^4.0.3",
-    "graphql-type-json": "^0.3.0",
-    "lodash": "^4.17.13",
-    "mkdirp": "^0.5.1",
-    "node-emoji": "^1.8.1",
-    "open": "^6.3.0",
-    "pg": "^7.7.1",
-    "pgtools": "^0.3.0",
-    "prettier": "^1.18.2",
-    "reflect-metadata": "^0.1.12",
-    "shortid": "^2.2.14",
-    "sqlite3": "^4.0.6",
-    "type-graphql": "^0.17.4",
-    "typedi": "^0.8.0",
-    "typeorm": "^0.2.18",
-    "typeorm-typedi-extensions": "^0.2.3",
-    "typescript": "^3.7.2"
-  },
   "devDependencies": {
-    "@types/express": "^4.16.0",
-    "@types/faker": "^4.1.5",
     "@types/jest": "^24.0.15",
-    "@types/supertest": "^2.0.8",
     "@typescript-eslint/eslint-plugin": "^2.2.0",
     "@typescript-eslint/parser": "^2.2.0",
-    "copyfiles": "^2.1.1",
-    "dotenv": "^8.0.0",
     "eslint": "^5.16.0",
     "eslint-config-prettier": "^5.0.0",
     "eslint-plugin-prettier": "^3.1.0",
-    "faker": "^4.1.0",
     "husky": "^1.3.1",
-    "jest": "^24.8.0",
+    "lerna": "^3.20.1",
     "lint-staged": "^8.1.0",
     "markdownlint-cli": "^0.19.0",
+    "prettier": "^1.18.2",
     "semantic-release": "^16.0.0-beta.31",
-    "supertest": "^4.0.2",
-    "ts-jest": "^24.0.2",
-    "ts-node": "^7.0.1"
-  },
-  "lint-staged": {
```

Comment:
Did these have to be moved out of package.json into rc files?

Reply:
At first, I've just removed the configs from `package.json` and created `.rc` files, but these files should be `.config.js` ideally. In monorepo setups, each package usually contains their required configuration for tooling. However, it is nice to have a common base configuration and, if needed, the package can extend/override the config. We can, for example, have a base Jest config file with `ts-jest` and coverage properties predefined and do something like this:

```javascript
// packages/core/jest.config.js
const baseConfig = require("../jest.base.config.js");
module.exports = {
  ...baseConfig,
  < custom properties >
}
```

I think it makes sense for `prettier` and `jest`. `lint-staged` would be the only tool configured in `package.json`, so I've also created a separate configuration file for it.